diff --git a/linux/sysroot_scripts/install-sysroot.py b/linux/sysroot_scripts/install-sysroot.py
index eaa1c2e..abd7eb4 100755
--- a/linux/sysroot_scripts/install-sysroot.py
+++ b/linux/sysroot_scripts/install-sysroot.py
@@ -155,7 +155,7 @@
   if sha1sum != tarball_sha1sum:
     raise Error('Tarball sha1sum is wrong.'
                 'Expected %s, actual: %s' % (tarball_sha1sum, sha1sum))
-  subprocess.check_call(['tar', 'mxf', tarball, '-C', sysroot])
+  subprocess.check_call(['tar', '--no-same-owner', '-mxf', tarball, '-C', sysroot])
   os.remove(tarball)
 
   with open(stamp, 'w') as s:
