diff --git a/v8/third_party/zlib/zutil.h b/v8/third_party/zlib/zutil.h
index e0466922..80e58c87 100644
--- a/v8/third_party/zlib/zutil.h
+++ b/v8/third_party/zlib/zutil.h
@@ -152,7 +152,7 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #  endif
 #endif
 
-#if defined(MACOS) || defined(TARGET_OS_MAC)
+#if defined(MACOS)
 #  define OS_CODE  7
 #  ifndef Z_SOLO
 #    if defined(__MWERKS__) && __dest_os != __be_os && __dest_os != __win32_os